parlai/core/torch_generator_agent.py

```diff
@@ -1611,7 +1714,22 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # get the actual word id from residual of the same division
         tok_ids = best_idxs % voc_size
 
-        return (hyp_ids, tok_ids, best_scores)
+        tok_scores = torch.gather(logprobs, 1, tok_ids.unsqueeze(0)).view(-1)
+
+        tok_ranks = (
+            logprobs.argsort(1, descending=True)
+            .argsort(1, descending=True)
```

Comment:
similarly here, the best indices need to be selected along the hypothesis ids

Reply:
Hm, I think this actually works as I did because `best_idx` gives the indices along the flattened concatenation of all of the beams in `logprobs`. It's not the same situation as what you pointed out with me using `tok_ids` when computing `tok_scores`, where the `tok_ids` represent indices along a single beam of probabilities. Since `best_idxs` represents indices along the concatenation of the beams in logprobs, we can first double arg sort logprobs along the dim 1 (along each beam), flatten the result, and then index by `best_idxs`. Am I missing something here?

Reply:
The fix is I need to not have `descending=True` flag in second `argsort`. I wrote some tests to check for errors like this.